.github/workflows/release.yml

```
@@ -1,40 +1,74 @@
+  packages: read
+  pull-requests: read
+
+concurrency:
+  group: release-${{ github.workflow }}-${{ github.run_id }}
+  cancel-in-progress: false
 
 jobs:
   release:
-    if: github.event.pull_request.merged == true && contains(github.event.pull_request.labels.*.name, 'release')
+    # Run only if Tests succeeded AND it was a push to main
+    if: >
```

actionlint (1.7.7)

if: condition "${{ github.event.workflow_run.conclusion == 'success' &&\n    github.event.workflow_run.event == 'push' &&\n    github.event.workflow_run.head_branch == 'main' }}\n" is always evaluated to true because extra characters are around ${{ }}

(if-cond)